### PR TITLE
Features/withdraw link api WIP

### PR DESCRIPTION
**Pull request by whatsadebugger**
Need some guidance as to how to finish this up. I'm very close I'm just missing a few pieces.

### CODE REVIEW

File: store/tx_manager.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ import (
 )
 
 const defaultGasLimit uint64 = 500000
+const withdrawFuncSelectorID string = "f3fef3a3"
```
Comment (se3000) [inline store/tx_manager.go]: Please add a comment documenting the actual function signature.

File: web/account_balance_controller_test.go [hunk 2]
```diff
@@ -49,3 +55,47 @@ func TestAccountBalanceController_Index(t *testing.T) {
...
 	assert.Equal(t, "0.000000000000000256", ab.LinkBalance.String())
 }
+
+func TestAccountBalanceController_Withdraw(t *testing.T) {
+	appWithAccount, cleanup := cltest.NewApplicationWithKeyStore()
+	defer cleanup()
+
+	ethMock := appWithAccount.MockEthClient()
+	sentAt := "0x5BA0"
+	hash := cltest.NewHash()
```
Comment (se3000) [inline web/account_balance_controller_test.go]: This is used later in the response to `sendRawTransaction` but that is thrown away, doesn't really matter here. The easiest way to know what the transaction hash is going to be is to set up an application with [cltest.NewConfigWithPrivateKey](https://github.com/smartcontractkit/chainlink/blob/master/integration/features_test.go#L48-L49). Once you have that, you should get the same expected value every time.

Comment (whatsadebugger) [reply web/account_balance_controller_test.go]:
```
	config, _ := cltest.NewConfigWithPrivateKey()
	app, cleanup := cltest.NewApplicationWithConfigAndKeyStore(config)
```

File: store/version.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package store
 
 // Version the version of application
-var Version string = "unset"
+var Version = "unset"
```
Comment (whatsadebugger) [inline store/version.go]: golint errors

File: web/account_balance_controller.go [hunk 4]
```diff
@@ -41,3 +45,33 @@ func (jsc *AccountBalanceController) Show(c *gin.Context) {
...
+func (abc *AccountBalanceController) Withdraw(c *gin.Context) {
+	store := abc.App.Store
+	txm := store.TxManager
+	wr := models.WithdrawalRequest{}
+	oa := common.HexToAddress("0x01")
+
+	if err := c.ShouldBindJSON(&wr); err != nil {
+		publicError(c, 400, err)
+	} else if wr.Amount.Cmp(naz) < 0 {
+		publicError(c, 400, errors.New("Must withdraw at least 0.000000000000000001 LINK"))
```
Comment (rupurt) [inline web/account_balance_controller.go]: This should be able to use the formatting from `Link#String()`

**Comment by rupurt** [comment]
@whatsadebugger cheers @NavyAdmiral. Exactly what I meant 😄

**Comment by whatsadebugger** [comment]
going to reopen this when im done

**Pull request closed without merging**